File: src/main/java/net/eduard/permissions/command/PermissionsGroupDeleteCommand.java

package net.eduard.permissions.command;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;

import net.eduard.api.lib.manager.CommandManager;
import net.eduard.permissions.EduPermissions;
import net.eduard.permissions.manager.PermissionsGroup;
import net.eduard.permissions.manager.PermissionsManager;

public class PermissionsGroupDeleteCommand extends CommandManager {

	public PermissionsGroupDeleteCommand() {
		super("delete", "deletar");
		setUsage("/permissions group delete <group>");
		setDescription("Deletar um grupo de permissões");
	}

	@Override
	public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {

		PermissionsManager manager = EduPermissions.getInstance().getManager();
		if (args.length < 3) {
			sendUsage(sender);
		} else {
			String nome = args[2];
			PermissionsGroup group = manager.getGroup(nome);
			if (group != null) {
				manager.removeGroup(group);
				sender.sendMessage(EduPermissions.getInstance().message("group-deleted").replace("$group", "" + nome));
			} else {
				sender.sendMessage(
						EduPermissions.getInstance().message("group-not-exists").replace("$group", "" + nome));
			}
//			
		}

		return true;
	}

}

File: src/main/java/net/eduard/permissions/command/PermissionsGroupDeleteCommand.kt
package net.eduard.permissions.command

import net.eduard.api.lib.command.Command
import net.eduard.api.lib.command.Sender
import net.eduard.permissions.EduPermissions
import net.eduard.permissions.api.PermissionsAPI
import net.eduard.permissions.core.PermMessages

class PermissionsGroupDeleteCommand :
    Command("delete", "deletar") {

    override fun onCommand(sender: Sender, args: List<String>) {
        val manager = PermissionsAPI.instance
        if (args.size < 3) {
            sendUsage(sender)
            return
        }
        val nome = args[2]
        val group = manager.getGroup(nome)
        if (group != null) {
            manager.unregisterGroup(group)
            sender.sendMessage(
                PermMessages.message("group-deleted")
                    .replace("\$group", "" + nome)
            )
            return
        }
        sender.sendMessage(
           PermMessages
                .message("group-not-exists")
                .replace("\$group", name))


    }


    init {
        usage = "/permissions group delete <group>"
        description = "Deletar um grupo de permissões"
    }
}